packages/blob/src/test/index.js

```diff
@@ -26,3 +26,31 @@ describe( 'getBlobTypeByURL', () => {
 		expect( getBlobTypeByURL() ).toBe( undefined );
 	} );
 } );
+
+describe( 'downloadBlob', () => {
+	const originalURL = window.URL;
+	const createObjectURL = jest.fn().mockReturnValue( 'blob:pannacotta' );
+	const revokeObjectURL = jest.fn().mockReturnValue( false );
+	beforeEach( () => {
+		window.URL = {
+			createObjectURL,
+			revokeObjectURL,
+		};
```

Comment:
Nit: Might be easier to spy on the global `URL` object?

Reply:
Actually, that's what I tried first but I discovered that these two static methods aren't available in the `global.URL()` constructor for some reason.

Unless I'm missing something?

I thought about extending it just for the tests, but it was easier to mock.

Reply:
I think the version of jsdom we use hasn't implement `URL.createObjectURL` and `URL.revokeObjectURL` yet. I believe we can just assign the Node.js's functions to them.


```js
import { URL } from 'url'; 
window.URL.createObjectURL = URL.createObjectURL; // Mock here if needed.
window.URL.revokeObjectURL = URL.revokeObjectURL; // Mock here if needed.
```

---

_Off topic:_

https://github.com/WordPress/gutenberg/blob/7d606b68ac856fda8916e8a80bab978cc080152d/packages/jest-preset-default/scripts/setup-globals.js#L5

This line is probably the reason why the global `URL` is overridden. I don't think it's relevant anymore as Node.js > v16 has the global URL available now.

Reply:
Thanks for digging deeper. 

>  I believe we can just assign the Node.js's functions to them

Do you see any advantage to doing that?

Reply:
Nope, it's just a nitpick that I was just curious why you chose to mock them completely instead 😆.